monkestation/code/modules/clothing/suits/reactive_armour.dm

```diff
@@ -0,0 +1,300 @@
+//Honk armor
+/obj/item/clothing/suit/armor/reactive/honk
+	name = "reactive honk armor"
+	desc = "An experimental suit of armor that honks violently."
+	reactivearmor_cooldown_duration = 10 SECONDS
+
+/obj/item/clothing/suit/armor/reactive/honk/hit_reaction(mob/living/carbon/human/owner, atom/movable/hitby, attack_text = "the attack", final_block_chance = 0, damage = 0, attack_type = MELEE_ATTACK)
+	if(!active)
+		return FALSE
+	if(prob(hit_reaction_chance))
+		if(world.time < reactivearmor_cooldown)
+			owner.visible_message("<span class='danger'>The horn is still recharging!</span>")
+			return FALSE
+		playsound(get_turf(owner),'sound/items/bikehorn.ogg', 100, 1)
+		owner.visible_message("<span class='danger'>[src] honks, converting the attack into a violent honk!</span>")
+		var/turf/owner_turf = get_turf(owner)
+		owner.Paralyze(3 SECONDS)
+		for(var/mob/living/carbon/target_atom as mob in ohearers(7, owner_turf))
+			target_atom.Paralyze(6 SECONDS)
+
+
+		reactivearmor_cooldown = world.time + reactivearmor_cooldown_duration
+		return TRUE
+
+
+/obj/item/clothing/suit/armor/reactive/honk/emp_act()
+	if(active)
+		active = FALSE
+		playsound(get_turf(src),'sound/items/bikehorn.ogg', 100, 1)
+		src.visible_message("<span class='danger'>[src] malfunctions, and honks extra hard!</span>")
+		for(var/mob/living/carbon/target_atom as mob in hearers(7, get_turf(src))) //Includes the person wearing it
+			target_atom.Paralyze(rand(5 SECONDS,20 SECONDS)) //Honk! :)
+	return
+
+//Mutation Armour
+/obj/item/clothing/suit/armor/reactive/mutation
+	name = "reactive mutation armor"
+	desc = "An experimental suit of armor that gives off radioactive waves."
+	var/list/possible_mutations = list()
+	reactivearmor_cooldown_duration = 30 SECONDS
+
+/obj/item/clothing/suit/armor/reactive/mutation/Initialize(mapload)
+	. = ..()
+	possible_mutations = GLOB.all_mutations
+	//These ones either don't let you unmutate, or are species specific
+	possible_mutations -= list(RACEMUT,CLUWNEMUT,MUTATE,ACIDOOZE,FIREBREATH,OVERLOAD)
+
+/obj/item/clothing/suit/armor/reactive/mutation/hit_reaction(mob/living/carbon/human/owner, atom/movable/hitby, attack_text = "the attack", final_block_chance = 0, damage = 0, attack_type = MELEE_ATTACK)
+	if(!active)
+		return FALSE
+	if(prob(hit_reaction_chance))
+		if(world.time < reactivearmor_cooldown)
+			owner.visible_message("<span class='danger'>The armour is still recharging!</span>")
+			return FALSE
+		playsound(get_turf(owner),'sound/effects/empulse.ogg', 100, 1)
+		owner.visible_message("<span class='danger'>[src] blocks [attack_text], sending out mutating waves of radiation!</span>")
+		var/turf/owner_turf = get_turf(owner)
+
+		for(var/mob/living/carbon/target_atom as mob in oviewers(7, owner_turf))
+			if(!ishuman(target_atom))
```

Comment:
what do you mean by this? I changed the variable in the for loop to carbon/human instead of just carbon, and I changed the second check to if(!istype(target_atom)), but I can't think of anything else that I could do.

Reply:
ok other than if there's something better I can do here, I'm done 🎉

Reply:
> what do you mean by this? I changed the variable in the for loop to carbon/human instead of just carbon, and I changed the second check to if(!istype(target_atom)), but I can't think of anything else that I could do.

Yes, that's what I meant.
A for loop with a specific typepath will only take the correct type